### PR TITLE
Feature/che 622 refactor token completion

### PR DESCRIPTION
**Pull request by EvansPie**
CHE-622

# What this PR does

Replace CreateClientTokenResponse to String for the client token

# Notable decisions & other stuff

n/a

# Before merging

_QA_

- [x] I manually tested this with a demo application (simulator)
- [ ] I manually tested this with a demo application (real device)
- [ ] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk

### CODE REVIEW

File: Sources/PrimerSDK/Classes/Data Models/PrimerSettings.swift [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #if canImport(UIKit)
 
-public typealias ClientTokenCallBack = (_ completionHandler: @escaping (Result<CreateClientTokenResponse, Error>) -> Void) -> Void
+public typealias ClientTokenCallBack = (_ completionHandler: @escaping (Result<String, Error>) -> Void) -> Void
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/Data Models/PrimerSettings.swift:3]: down the line, maybe we wrap this in a struct something? (keeping in mind the feedback regarding decodable etc.). Seems like an easy way to introduce bugs passing an anonymous strings around.

File: Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift [hunk 2]
```diff
@@ -82,25 +82,16 @@ extension String {
...
-                    } catch {
-                        log(logLevel: .error, title: "PARSER", message: "Failed to parse client token", prefix: nil, suffix: nil, bundle: Bundle.primerFrameworkIdentifier, file: #file, className: nil, function: #function, line: #line)
-                    }
-                }
-            }
-
-        }
-        return DecodedClientToken()
+    
+    var decodedJWTToken: DecodedClientToken? {
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift:86]: is this tested?

**Review by ceriksson — APPROVED** [review]
looks fine, may wanna double check the new base 64 decoding works properly.

File: Sources/PrimerSDK/Classes/Data Models/PrimerSettings.swift (hunk 1 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Data Models/PrimerSettings.swift:3]: Well, it's a very usual practice to pass primitive vars if possible. This is what Mayur wanted, to not have to create a custom decodable object just as a string wrapper. Moreover, now we don't even have the expiration timestamp in this object, so it'd be a struct containing just a string.

File: Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift (hunk 2 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift:86]: Yeah, quite a lot actually. It's one of the snippets that I was using on my previous company. I'd need to write some test cases though. I do think however that I should change the name, cause this isn't the whole token information, just the payload. @xevious78 @ceriksson would we need any other information from the token? Maybe I should ask Will about it

**Pull request merged**